### DISCUSSION TITLE
Discrepancy in the recommended

### DISCUSSION THREAD

**Discussion by erosman** [eslint/config-inspector#15] [Q&A]
_eslint@9.0.0-rc.0
ESLint Config Inspector v0.2.1_

While checking **ESLint Config Inspector**, I noticed some discrepancies and I am not sure whether they belong to the Config Inspector or the ESLint runtime.

The config that I use imports `configs.recommended` e.g.

```js
import js from "@eslint/js";

export default [
  js.configs.recommended, 
  // ...
];
```

There are 4 unused recommended rules.

![image](https://github.com/eslint/eslint/assets/6458876/d592c1c3-5dbd-416a-a625-7a3878b676ab)

The `recommenced` data comes from the runtime.
_(See also: [Mark recommended](https://github.com/eslint/config-inspector/issues/6), [Mark imported config](https://github.com/antfu/eslint-flat-config-viewer/issues/19))_

There appears to be a discrepancy somewhere since they don't appear in `js/src/configs/eslint-recommended.js`.

https://github.com/eslint/eslint/blob/b7cf3bd29f25a0bab4102a51029bf47c50f406b5/packages/js/src/configs/eslint-recommended.js#L11-L75

There are also a few rules listed under in `config #1` which is `configs.recommended` that dont have `recommended: true` e.g.

![image](https://github.com/eslint/eslint/assets/6458876/ea9fea76-33fd-4a2b-8338-4145d41cddba)

**Comment by nzakas** [comment]
@antfu might be able to answer this better.

**Comment by antfu** [comment]
@erosman I guess you have a different versions of `eslint` and `@eslint/js`?

**Comment by erosman** [comment]
How can I check the version/content of `@eslint/js`?

Looking at `/home/***/node_modules/@eslint/js/src/configs/eslint-recommended.js`, it matches [/packages/js/src/configs/eslint-recommended.js](https://github.com/eslint/eslint/blob/b7cf3bd29f25a0bab4102a51029bf47c50f406b5/packages/js/src/configs/eslint-recommended.js)

Can there be an issue generated from VS Code ESLint extension? :thinking: 

_eslint@9.0.0-rc.0
ESLint Config Inspector v0.2.1
VS Code v1.87.2
VS Code ESLint extension v2.4.4_

**Reply by antfu** [reply]
Try `pnpm why -r @eslint/js`

**Reply by erosman** [reply]
This is I what I get.

```
npm why -r @eslint/js

@eslint/js@9.0.0-rc.0 dev
node_modules/@eslint/js
  @eslint/js@"9.0.0-rc.0" from eslint@9.0.0-rc.0
  node_modules/eslint
    dev eslint@"^9.0.0-rc.0" from the root project

@eslint/js@8.57.0 dev peer
node_modules/eslint/node_modules/eslint/node_modules/@eslint/js
  @eslint/js@"8.57.0" from eslint@8.57.0
  node_modules/eslint/node_modules/eslint
    peer eslint@"^6.0.0 || ^7.0.0 || >=8.0.0" from @eslint-community/eslint-utils@4.4.0
    node_modules/eslint/node_modules/@eslint-community/eslint-utils
      @eslint-community/eslint-utils@"^4.2.0" from eslint@9.0.0-rc.0
      node_modules/eslint
        dev eslint@"^9.0.0-rc.0" from the root project
      @eslint-community/eslint-utils@"^4.2.0" from eslint@8.57.0
```

**Reply by antfu** [reply]
Clearly you have `@eslint/js@8.57.0`

**Reply by erosman** [reply]
What shall I do?
I am new to ESLint & npm.

**Comment by mdjermanovic** [comment]
I'd guess that `config-inspector` gets builtin rules from an ESLint version that is installed as its peer dependency rather than from locally installed prerelease version.

https://github.com/eslint/config-inspector/blob/509a1ae3c661abdb4a0a89df7bc305746bab855a/package.json#L32-L34

https://github.com/eslint/config-inspector/blob/509a1ae3c661abdb4a0a89df7bc305746bab855a/src/configs.ts#L31
